network/src/metrics.rs

```diff
@@ -0,0 +1,139 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use prometheus::{default_registry, register_int_gauge_vec_with_registry, IntGaugeVec, Registry};
+use std::sync::Arc;
+
+pub trait NetworkMetrics {
```

Comment:
Ideally I would use only one struct to report the metrics both for worker and primary nodes as those would normally be deployed as separate nodes. However, currently the worker & primary are deployed as part of the same binary in SUI and this would lead to re-registering the same metric (which would lead to an error). So had to split those but to minimise the implementation I used a common interface .